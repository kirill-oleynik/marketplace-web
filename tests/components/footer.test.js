const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const { Footer } = require('../../app/components/footer');

const onTermsOfUseClickSpy = jest.fn();
const onPrivacyPolicyClickSpy = jest.fn();

const renderFooter = () => (
  <Footer
    t={(translation) => translation}
    onTermsOfUseClick={onTermsOfUseClickSpy}
    onPrivacyPolicyClick={onPrivacyPolicyClickSpy}
  />
);

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      renderFooter()
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('on Terms Of Use click', () => {
  it('calls onTermsOfUseClick', () => {
    const component = shallow(
      renderFooter()
    );

    component.find('.main-footer__terms-link').simulate('click');

    expect(onTermsOfUseClickSpy).toHaveBeenCalled();

    onTermsOfUseClickSpy.mockReset();
    onTermsOfUseClickSpy.mockRestore();
  });
});

describe('on Privacy Policy click', () => {
  it('calls onPrivacyPolicyClick', () => {
    const component = shallow(
      renderFooter()
    );

    component.find('.main-footer__privacy-link').simulate('click');

    expect(onPrivacyPolicyClickSpy).toHaveBeenCalled();

    onPrivacyPolicyClickSpy.mockReset();
    onPrivacyPolicyClickSpy.mockRestore();
  });
});
