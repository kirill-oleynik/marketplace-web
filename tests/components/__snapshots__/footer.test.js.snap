// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<footer>
  <Container
    tag="div"
  >
    <div
      className="main-footer main-footer-js"
    >
      <div
        className="main-footer__wrap"
      >
        <Link
          href="/"
        >
          <a
            className="page-main-logo in-blue-500"
          >
            footer.applicationName
          </a>
        </Link>
        <p
          className="font-14 in-black-035 ml-10 mb-0"
        >
          footer.copyright
        </p>
      </div>
      <div
        className="main-footer__wrap"
      >
        <span
          className="main-footer__link main-footer__terms-link"
          onClick={[Function]}
        >
          footer.termsOfUse
        </span>
        <span
          className="main-footer__link main-footer__privacy-link"
          onClick={[Function]}
        >
          footer.privacyPolicy
        </span>
      </div>
    </div>
  </Container>
</footer>
`;
